addons/account_audit_trail/models/account_invoice.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_account_audit_trail_except_once_post(self):
+        if any(m.posted_before and m.company_id.check_account_audit_trail for m in self) and not self._context.get('force_delete'):
```

Comment:
```suggestion
        if not self._context.get('force_delete') and any(
            move.posted_before and move.company_id.check_account_audit_trail 
            for move in self
        ):
```

Reply:
Note that the context is easily bypass-able...
Note sure how it works regarding legal stuff